internal/util/util.go

```diff
@@ -175,6 +175,9 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 		// bound to the existing snapshot.
 		if snapshotContent.Status == nil || snapshotContent.Status.SnapshotHandle == nil {
 			log.Infof("Waiting for volumesnapshotcontents %s to have snapshot handle. Retrying in %ds", snapshotContent.Name, interval/time.Second)
+			if snapshotContent.Status == nil && snapshotContent.Status.Error != nil {
```

Comment:
Looks like the if condition check should be 
`if snapshotContent.Status != nil && snapshotContent.Status.Error != nil`